src/repart/repart.c

```diff
@@ -7250,6 +7250,82 @@ static int resolve_copy_blocks_auto_candidate(
         return true;
 }
 
+static int resolve_copy_blocks_auto_candidate_harder(
+                dev_t start_devno,
+                GptPartitionType partition_type,
+                dev_t restrict_devno,
+                dev_t *ret_found_devno,
+                sd_id128_t *ret_uuid) {
+
+        _cleanup_(sd_device_unrefp) sd_device *d = NULL, *nd = NULL;
+        int r;
+
+        /* A wrapper around resolve_copy_blocks_auto_candidate(), but looks for verity/verity-sig associated
+         * partitions, too. i.e. if the input is a data or verity partition, will try to find the
+         * verity/verity-sig partition for it, based on udev metadata. */
+
+        const char *property;
+        if (partition_verity_to_data(partition_type.designator) >= 0)
+                property = "ID_DISSECT_PART_VERITY_DEVICE";
+        else if (partition_verity_sig_to_data(partition_type.designator) >= 0)
```

Comment:
Please introduce `partition_designator_is_verity()` and use it here together with `partition_designator_is_verity_sig()`. That's much more clear than reusing `partition_verity_to_data()` and `partition_verity_sig_to_data()` for the same purpose.

Feel free to reimplement `partition_designator_is_verity_sig()` on top of `partition_verity_sig_to_data()` though.